control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -247,8 +254,18 @@ func (r *AWSEndpointServiceReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	// Fetch the HostedControlPlane
+	hcpList := &hyperv1.HostedControlPlaneList{}
+	if err := r.List(ctx, hcpList, &client.ListOptions{Namespace: req.Namespace}); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get resource: %w", err)
+	}
+	if len(hcpList.Items) != 1 {
```

Comment:
Why do we check for 0 in https://github.com/openshift/hypershift/pull/710/files#diff-6995c532d3cc9b0428062dc5d76589196d62dc18751b385d26dcc4d035ea4f1aR123 and do something different here?

Reply:
I think you are right.  We should probably do the same thing in both cases.